### PR TITLE
Fix pop up on expand animation.

### PR DESCRIPTION
**Pull request by oyj9109**
Context: on expansion, immediate style change is queued and then transition animation begins. They could collide within the same animation frame which resulted in pop ups (refer to the below screencasts). 
https://github.com/subscriptions-project/swg-js/blob/02339a41383c78d01c3f9b2bbc1249f553d15ca9/src/components/dialog.js#L504-L514

Calling `requestAnimationFrame` to make sure they are executed in the expected time sequence.

Before: https://screencast.googleplex.com/cast/NTg0NDI5OTEyNTA5NjQ0OHw2YWVjMGEwOC0xZA
After: https://screencast.googleplex.com/cast/NTc3MDE0NzE4NTgxOTY0OHxiM2E4YjdhZi1jZA

### CODE REVIEW

**Review by fikus — APPROVED** [review]

**Comment by oyj9109** [comment]
Added availability check in case some browser doesn't support the function. (Refer to https://caniuse.com/?search=requestAnimationFrame)

File: src/components/dialog.js [hunk 1]
```diff
@@ -504,14 +504,23 @@ export class Dialog {
...
+            transition(
+              this.getElement(),
+              {
+                'transform': this.getDefaultTranslateY_(),
+              },
+              300,
+              'ease-out'
+            );
+          };
+          if (typeof requestAnimationFrame === 'function') {
```
Comment (ChrisAntaki) [inline src/components/dialog.js:517]: So I think this check could be removed. Swgjs' supported browsers have the `requestAnimationFrame` method

Comment (oyj9109) [reply src/components/dialog.js]: Oh cool, good to know. Reverted the last commit.

Comment (ChrisAntaki) [reply src/components/dialog.js]: Thanks!

**Pull request merged**